File: code_examples/Multi-Platform/src12/c/set12.c
void calc_divv(int np, int nlev, int nelem, double *ru) {
	// size and intent of array arguments for f2py
	// ru :: np*np*(nlev+1)*nelem, inout
	
	int idx;

	for (idx=0; idx<np*np*(nlev+1)*nelem; idx++) {
		ru[idx] = 1.2;
	}
}

File: code_examples/Multi-Platform/src12/f90/set12.f90

SUBROUTINE calc_divv(np, nlev, nelem, ru)
  IMPLICIT NONE
  INTEGER, INTENT(IN) :: np, nlev, nelem
  REAL(8), DIMENSION(np,np,nlev+1,nelem), INTENT(INOUT) :: ru

  INTEGER :: ie, k

  DO ie=1,nelem
    DO k=1,nlev+1
      ru(:,:,k,ie) = 1.2D0
    END DO
  END DO
END SUBROUTINE
    
